src/cli/Secrets.ts

```diff
@@ -245,7 +245,7 @@ function makeSecretEnvCommand() {
           throw Error(`Error when retrieving secret: ${err.message}`);
         }
         try {
-          const shellPath = process.env.SHELL ?? "sh";
+          const shellPath = process.env.SHELL ?? 'sh';
```

Comment:
With such a big program, I would standardize all uses of environment variables to the `src/bin/config.ts`.

This makes it easier to mock the code as well. Since you're not using side-effects everywhere in the codebase.

Also that's what we have to do in Haskell. Good to standardize patterns.

Also the official path is actually `/bin/sh`, that's what exists in the NixOS. As well as `/usr/bin/env`. But `/bin/sh` is usually assumed to exist.

Consider what happens if the shell doesn't exist. You need to handle that error. And report a proper error under STDERR as well as a failed exit code from the sysexits standard.